### ISSUE TITLE
Expand Matrix URI scheme actions to include action=knock

### ISSUE THREAD

**Issue by Mikaela** [matrix-org/matrix-spec#1015]

**Comment by ShadowJonathan** [comment]
Wouldn’t this be covered under `action=join`? 

Couldn’t that spec segment be reworded to include “knock” as a valid action?

(cc @turt2live and @KitsuneRal)

**Comment by KitsuneRal** [comment]
Yes, and it takes a pretty small straightforward MSC, before adding it to the spec text. How about drafting one? :)

**Comment by ShadowJonathan** [comment]
Sounds good, I talked to @Mikaela and Nico (from Nheko) about it, and so I'll make that MSC

**Comment by Mikaela** [comment]
Would this also be reflected in matrix.to handling or does it need a different issue?

**Comment by KitsuneRal** [comment]
matrix.to predates Matrix URIs and doesn't abide by their conventions. You'll have to submit a separate issue at its repo.

**Comment by turt2live** [comment]
the spec for matrix.to exists on our end though? The project's support is of course a different issue, but the specification for what it should do is needed here.

**Comment by Mikaela** [comment]
I am having difficulties getting to my mindset 8 days ago on what exactly is the issue on matrix.to side considering it just attempts to open clients and offers commands. Should it be aware of join_rule and offer /knock for Nheko users instead of join? Won't clients just offee knocking if join fails as per #3780? If not, isn't that client rather than matrix.to issue or even another spec issue?

**Comment by turt2live** [comment]
it should just work with the clients, but if matrix.to is translating the link into something for the client to open with then matrix.to needs to be aware of how to do that for the client.
